File: FmWk/Assets/MyFramework/Modules/SceneModule/Script/SceneControl/ISceneLoadCompBase.cs
using System;
using System.Collections;

namespace MyFramework
{
    /// <summary>
    /// 场景控制类接口
    /// </summary>
    public interface ISceneLoadCompBase
    {
        /// <summary>
        ///获取场景切换进入
        /// </summary>
        /// <returns></returns>
        float GetProcess();

        /// <summary>
        /// 获取场景名称
        /// </summary>
        /// <returns></returns>
        string GetSceneName();

        /// <summary>
        /// 场景加载
        /// </summary>
        IEnumerator LoadScene();

        /// <summary>
        /// 场景卸载
        /// </summary>
        IEnumerator UnloadScene();
    }
}

File: FmWk/Assets/MyFramework/Modules/SceneModule/Lua/ISceneLoadCompBase.lua
ISceneLoadCompBase = Class.define("ISceneLoadCompBase",BaseModelComp);

function ISceneLoadCompBase:Load(...)
	self.Process = 0;
	self:super(ISceneLoadCompBase,"Load",...);
end

--场景加载进度
function ISceneLoadCompBase:GetProcess()
   return self.Process
end

--场景名称
function ISceneLoadCompBase:GetSceneName()
   return "" 
end 


function ISceneLoadCompBase:LoadScene()
end


function ISceneLoadCompBase:UnloadScene()
end 
